test/serverExplorer.test.ts

```diff
@@ -0,0 +1,193 @@
+        id: 'type',
+        visibleName: 'the type'
+    };
+
+    const serverHandle: Protocol.ServerHandle = {
+        id: 'id',
+        type: serverType
+    };
+
+    const stateChange: Protocol.ServerStateChange = {
+        server: serverHandle,
+        state: 1
+    };
+
+    const ProcessOutput: Protocol.ServerProcessOutput = {
+        processId: 'process id',
+        server: serverHandle,
+        streamType: 0,
+        text: 'the type'
+    };
+
+    const findServerBeans = {
+        length: 1,
+        fullVersion: 'version',
+        location: 'path',
+        name: 'EAP',
+        serverAdapterTypeId: 'org.jboss',
+        specificType: 'EAP',
+        typeCategory: 'EAP',
+        version: '7.1'
+    };
+
+    const status = {
+        code: 0,
+        message: 'ok',
+        pluginId: 'unknown',
+        severity: 0
+    };
+
+    test('InsertServer call should add server to tree data model', () => {
+        const refreshStub = sandbox.stub(serverExplorer, 'refresh').returns(null);
+        const getChildren = serverExplorer.getChildren(null);
+        serverExplorer.insertServer(serverHandle);
+        expect(refreshStub).calledOnce;
+        expect(getChildren).deep.equals(undefined);
+    });
+
+    test('removeServer call should remove server from tree data model', () => {
+        const disposeStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+            dispose: () => {}
+        });
+        const getChildren = serverExplorer.getChildren(null);
+        sandbox.stub(serverExplorer, 'refresh');
+        serverExplorer.removeServer(serverHandle);
+        expect(disposeStub).calledOnce;
+        expect(getChildren).deep.equals(undefined);
+    });
+
+    test('serverExplorer.updateServer call should update server state to received in state change event', () => {
+        const serviceStub = sandbox.stub(serverExplorer.servers, 'get').returns({
+            stateChange
+        });
+        const clearStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {
+                return true;
+            }
+        });
+        sandbox.stub(serverExplorer, 'refresh');
+        const updateserver = serverExplorer.updateServer(stateChange);
+        expect(serviceStub).calledOnce;
+        expect(clearStub).calledOnce;
+        expect(clearStub).calledOnceWith(updateserver);
+    });
+
+    test('serverExplorer.showOutput call should show servers output channel', () => {
+        const showStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {}
+        });
+        serverExplorer.showOutput(serverHandle);
+        expect(showStub).calledOnce;
+
+    });
+
+    test('serverExplorer.getTreeItem call should return TreeItem instance with corresponding label', () => {
+        const serverHandleStub = sandbox.stub(serverExplorer.serverStatus, 'get').resolves({
```

Comment:
wrong. Just call getTreeItem with a ServerHandle instance and verify returned tree item contains all required text.